src/osp/Active/SysNewton.cpp

```diff
@@ -393,6 +455,45 @@ void SysNewton::body_apply_torque(ACompRigidBody_t &body, Vector3 torque) noexce
     body.m_netTorque += torque;
 }
 
+Vector4 SysNewton::compute_body_CoM(ActiveScene& rScene,
+    ActiveEnt root, Matrix4 currentTransform)
+{
+    ActiveEnt nextChild = root;
+    Vector3 localCoM{0.0f};
+    float localMass = 0.0f;
+
+    while (nextChild != entt::null)
+    {
+        auto const &childHeir = rScene.reg_get<ACompHierarchy>(nextChild);
+        auto const &childTransform = rScene.reg_get<ACompTransform>(nextChild);
+
+        Matrix4 childMatrix = currentTransform * childTransform.m_transform;
+        auto* massComp = rScene.get_registry().try_get<ACompMass>(nextChild);
+
+        if (massComp)
+        {
+            float childMass = rScene.reg_get<ACompMass>(nextChild).m_mass;
+
+            Vector3 offset = childMatrix.translation();
+
+            localCoM += childMass * offset;
+            localMass += childMass;
+
+        }
+
+        Vector4 subCoM = compute_body_CoM(rScene, childHeir.m_childFirst, childMatrix);
+        localCoM += subCoM.w() * subCoM.xyz();
+        localMass += subCoM.w();
+
+        nextChild = childHeir.m_siblingNext;
+    }
+    if (localMass > 0.0f)
```

Comment:
how could you get neg mass?

Reply:
You can't, I'm just checking for nonzero mass, and have an aversion to direct float comparisons

Reply:
You should have an aversion to direct float comparisons 

Document in the code?